src/froll.c

```diff
@@ -397,6 +397,277 @@ void frollsumExact(double *x, uint64_t nx, ans_t *ans, int k, double fill, bool
   }
 }
 
+/* fast rolling max */
+void frollmax(unsigned int algo, double *x, uint64_t nx, ans_t *ans, int k, int align, double fill, bool narm, int hasna, bool verbose) {
+  if (nx < k) {
+    if (verbose)
+      snprintf(end(ans->message[0]), 500, _("%s: window width longer than input vector, returning all %f vector\n"), __func__, fill);
```

Comment:
Maybe better for follow-up:

Wouldn't the expected behavior here be to just set `k=nx` instead & proceed?

Why should this have different output (pseudocode):

```r
x = c(1, 4, 5, 6, 2, 3, 4)
frollmax(x, k = 7) # right-aligned
# 1 4 5 6 6 6 6
frollmax(x, k = 8) # what I'd expect
# 1 4 5 6 6 6 6
```

Reply:
Oh, I mis-read a bit. I see, it does give the same output, namely `rep(fill, length(x))` for both cases. CMIIW, resolving for now.